pkg/workload/workload.go

```diff
@@ -520,3 +524,40 @@ func RemoveFinalizer(ctx context.Context, c client.Client, wl *kueue.Workload) e
 	}
 	return nil
 }
+
+// AdmissionChecksForWorkload returns AdmissionChecks that should be assigned to a specific Workload based on
+// ClusterQueue configuration and ResourceFlavors
+func AdmissionChecksForWorkload(log *logr.Logger, wl *kueue.Workload, cq *kueue.ClusterQueue) sets.Set[string] {
+	if len(cq.Spec.AdmissionChecks) != 0 {
+		return sets.New(cq.Spec.AdmissionChecks...)
+	}
+
+	// Kueue sets AdmissionChecks first based on ClusterQueue configuration and at this point Workload has no
+	// ResourceFlavors assigned, so we cannot match AdmissionChecks to ResourceFlavor.
+	// After Quota is reserved, another reconciliation happens and we can match AdmissionChecks to ResourceFlavors
+	if wl.Status.Admission == nil {
+		log.V(2).Info("Workload has no Admission", "Workload", klog.KObj(wl))
+		return nil
+	}
+
+	assignedFlavors := sets.New[string]()
+	for _, podSet := range wl.Status.Admission.PodSetAssignments {
+		for _, flavor := range podSet.Flavors {
+			assignedFlavors.Insert(string(flavor))
```

Comment:
```suggestion
	var assignedFlavors []kueue.ResourceFlavorReference
	for _, podSet := range wl.Status.Admission.PodSetAssignments {
		for _, flavor := range podSet.Flavors {
			assignedFlavors = append(assignedFlavors, flavor)
```

If my previous comment brought you any confusion, sorry, but I think that here is slice should be enough.

Reply:
Fixed